src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -57,6 +48,19 @@ private static void ThreadStart_Context(object state)
         internal void ThreadStart(object obj)
         {
             _startArg = obj;
+
+            if (_startCulture != null)
+            {
+                CultureInfo.CurrentCulture = _startCulture;
+                _startCulture = null;
+            }
+
+            if (_startUICulture != null)
+            {
+                CultureInfo.CurrentUICulture = _startUICulture;
+                _startUICulture = null;
+            }
```

Comment:
Does the `ThreadStart()` constructor needs the same?

Reply:
Oh, right, yes.